include/dwfl_wrapper.hpp

```diff
@@ -34,7 +35,7 @@ struct DwflWrapper {
   DwflWrapper(const DwflWrapper &other) = delete;            // avoid copy
   DwflWrapper &operator=(const DwflWrapper &other) = delete; // avoid copy
 
-  DDRes attach(pid_t pid, const Dwfl_Thread_Callbacks *callbacks,
+  DDRes attach(pid_t pid, const UniqueElf &ref_elf, bool avoid_new_attach,
```

Comment:
Why add a new parameter `avoid_new_attach` that if true will always result in no attach and `ddres_warn(DD_WHAT_UW_MAX_PIDS)` ?